dockerfiles/Dockerfile

```diff
@@ -12,11 +15,25 @@ RUN gzip -d flyway-commandline-${FLYWAY_VERSION}.tar.gz \
   && chmod -R a+r /flyway \
   && chmod a+x /flyway/flyway
 
+##################################################
+# Step 1 - Build production flyway image
+##################################################
+FROM eclipse-temurin:17-jre-jammy as flyway
+
+WORKDIR /flyway
+
+ARG FLYWAY_VERSION
+
+COPY --from=unpack /flyway /flyway
```

Comment:
> [!Note]
> All that the original steps did was copy the `.tar.gz` into the container, unpack it, and set permissions on it.  We can simply do all of that in a build stage and then copy the resulting files into this container to avoid the cost of the extra layers.